test/e2e-sno-disruptive/sno_disruptive_test.go

```diff
@@ -1,7 +1,261 @@
+		if err == nil {
+			t.Logf("the static pod monitor: %v is running since: %v", monitorPod.Name, monitorPod.CreationTimestamp)
+			return false, nil /*retry*/
+		} else if !errors.IsNotFound(err) {
+			t.Logf("error while getting kube-apiserver-startup-monitor: %v", err)
+			return false, nil /*retry*/
+		}
+
+		kasPods, err := cs.Kube.CoreV1().Pods("openshift-kube-apiserver").List(context.TODO(), metav1.ListOptions{LabelSelector: "apiserver=true"})
+		if err != nil {
+			t.Logf("error while getting kube-apiserver pods: %v", err)
+			return false, nil /*retry*/
+		}
+		if len(kasPods.Items) == 0 {
+			t.Log("expected to find the kube-apiserver static pod but haven't found any")
+			return false, nil /*retry*/
+		}
+		for _, kasPod := range kasPods.Items {
+			if fallbackFor, ok := kasPod.Annotations[annotations.FallbackForRevision]; ok {
+				t.Logf("%v pod has %v annotation set to: %v", kasPod.Name, annotations.FallbackForRevision, fallbackFor)
+				return false, nil /*retry*/
+			}
+		}
+		// TODO: on an HA cluster we could also check if pods are on the same revision
+
+		ckaso, err := cs.Operator.Get(context.TODO(), "cluster", metav1.GetOptions{})
+		if err != nil {
+			t.Logf("unable to get kube-apiserver-operator resource: %v", err)
+			return false, nil /*retry*/
+		}
+		if !v1helpers.IsOperatorConditionFalse(ckaso.Status.Conditions, "StaticPodFallbackRevisionDegraded") {
+			t.Logf("StaticPodFallbackRevisionDegraded condition is set")
+			return false, nil /*retry*/
+		}
+
+		if time.Since(startTs) > mustBeReadyFor {
+			t.Logf("The cluster has been in good condition for %s", mustBeReadyFor.String())
+			return true, nil /*done*/
+		}
+		return false, nil /*wait a bit more*/
+	})
+}
+
+func setUnsupportedConfig(t testing.TB, cs clientSet, cfg map[string][]string) {
+	t.Helper()
+
+	t.Logf("Setting UnsupportedConfigOverrides to %v", cfg)
+	err := retry.OnError(retry.DefaultRetry, func(error) bool { return true }, func() error {
+		raw, err := json.Marshal(cfg)
+		if err != nil {
+			t.Log(err)
+			return err
+		}
+		ckaso, err := cs.Operator.Get(context.TODO(), "cluster", metav1.GetOptions{})
+		if err != nil {
+			t.Log(err)
+			return err
+		}
+		ckaso.Spec.UnsupportedConfigOverrides.Raw = raw
+		_, err = cs.Operator.Update(context.TODO(), ckaso, metav1.UpdateOptions{})
+		if err != nil {
+			t.Log(err)
+		}
+		return err
+	})
+	require.NoError(t, err)
+}
+
+func waitForFallbackDegradedCondition(t testing.TB, cs clientSet) {
+	t.Helper()
+
+	t.Log("Waiting for StaticPodFallbackRevisionDegraded condition, interval = 20s, timeout = 6min")
+	err := wait.Poll(20*time.Second, 6*time.Minute, func() (bool, error) {
+		ckaso, err := cs.Operator.Get(context.TODO(), "cluster", metav1.GetOptions{})
+		if err != nil {
+			t.Logf("unable to get kube-apiserver-operator resource: %v", err)
+			return false, nil /*retry*/
+		}
+
+		if v1helpers.IsOperatorConditionFalse(ckaso.Status.Conditions, "StaticPodFallbackRevisionDegraded") {
```

Comment:
why not check for True and return true? Seems to be more direct in a loop like this.